apps/backend/src/applications/applications.controller.ts

```diff
@@ -38,6 +39,36 @@ export class ApplicationsController {
     return await this.applicationsService.submitApp(application, user);
   }
 
+  @Post('/decision')
+  @UseGuards(AuthGuard('jwt'))
+  async makeDecision(
+    @Body('applicantId', ParseIntPipe) applicantId: number,
+    @Body('decision') decision: 'ACCEPT' | 'REJECT',
+    @Request() req,
+  ): Promise<void> {
+    //Authorization check for admin and recruiters
+    if (![UserStatus.ADMIN, !UserStatus.RECRUITER].includes(req.user.status)) {
+      throw new UnauthorizedException();
+    }
+
+    //Check if the string decision matches with the Decision enum
+    const decisionEnum: Decision = Decision[decision as keyof typeof Decision];
```

Comment:
❓We definitely need to make sure that the `decision` we get is a valid `Decision` but I _think_ line 46 in this file (where we type the `decision` parameter) tells Nest to check this for us and throw an error if the type doesn't match. Can you try commenting out this check and passing in an invalid value as the `decision` and see if the endpoint still responds with a `400 Bad Request` error?

Reply:
⛏️A potentially better alternative altogether is to make an application decision dto whose fields are `applicantId` and `decision.` Then, you can add decorators from `class-validator` such as `@IsEnum(Decision)` to the fields to ensure that Nest performs type checks at runtime.